PWGHF/D2H/Tasks/taskDs.cxx

```diff
@@ -170,11 +250,29 @@ struct HfTaskDs {
 
   /// Fill histograms of quantities for the KKPi daugther-mass hypothesis
   /// \param candidate is candidate
-  template <typename T1>
+  template <bool useMl, bool fillCent, typename T1>
   void fillHistoKKPi(const T1& candidate)
   {
     auto pt = candidate.pt();
-    registry.fill(HIST("hMass"), hfHelper.invMassDsToKKPi(candidate), pt);
+    if constexpr (useMl) {
+      std::vector<float> outputMl = {-999., -999., -999.};
+      if (candidate.mlProbDsToKKPi().size() > 0) {
```

Comment:
This should be always true if the right selection is applied in the filter using `doprocessMcWithMl`. So in principle there is no need to check